### DISCUSSION TITLE
How to set up YARP as public facing app for multiple applications/destinations

### DISCUSSION THREAD

**Discussion by lindsve** [dotnet/yarp#1423] [Q&A] [answered]
I'm trying to set up a monitoring solution for our applications using Grafana, Prometheus and Loki and my idea is to use an Azure App Service (due to integrated SSL) with YARP as the public facing application for this.

I initially tried to map different ports to different destinations (worked locally):
:443 -> Grafana
:9090 -> Prometheus
:3100 -> Loki

But then I finally figured out that App Service only exposes/listens to a single port.

I'm now trying to manipulate the url paths to get this working:
mysite.azurewebsites.com -> Grafana (http://10.0.1.1:8080)
mysite.azurewebsites.com/prometheus -> Prometheus (http://10.0.1.2:9090)
mysite.azurewebsites.com/loki -> Loki (http://10.0.1.3:3100)

The missing piece now is that I don't want the `/prometheus/`-part in the url to be included in the request to the backend destination. I tried to remove it with the PathRemovePrefix transform, but then that part is also removed on the public end of the request and I end up in Grafana instead of Prometheus.

What I want is:
Enter url in the browser: mysite.azurewebsites.com/prometheus/targets
The prometheus destination receives a request for mysite.azurewebsites.com/targets and shows the output from /targets while my browser stays at the /prometheus/targets address.

/prometheus and /loki is added to the urls here just for route matching. Other suggestions for routing traffic to the different destinations are welcome.

What am I missing here?

**Comment by Kahbazi** [comment, marked as answer]
You need to add [path remove prefix](https://microsoft.github.io/reverse-proxy/articles/transforms.html#pathremoveprefix) transform to your routes.

```
{ "PathRemovePrefix": "/prometheus" }
```

**Reply by lindsve** [reply]
I tried this, but it didn't work. But I thought the issue was YARP, but it's Prometheus as I forgot to update the root/external url.

**Comment by 0xStuart** [comment]
I'm trying to proxy a number of API ports on a remote server.
```
0.0.0.0:5001  =>  RemoteHost.com:7001
0.0.0.0:5002  =>  RemoteHost.com:7002
0.0.0.0:5003  =>  RemoteHost.com:7003
0.0.0.0:5004  =>  RemoteHost.com:7004
```

YARP seems to deal with URL mapping, and not port mapping.  What am I missing?

**Reply by Tratcher** [reply]
You can route by port using the Hosts field
```json
        "Match": {
          "Hosts" : [ "*:5001" ]
```
